test/hotspot/jtreg/serviceability/jvmti/events/NotifyFramePopStressTest/libNotifyFramePopStressTest.cpp

```diff
@@ -0,0 +1,206 @@
+  name = get_method_name(jvmti, jni, method);
+  LOG("FramePop(%d) event from method: %s %s\n", popCount, csig, name);
+
+  if (strcmp(name, "main") != 0) { // ignore FRAME_POP for main that comes in as the test exits
+    if (strcmp(name, lastNotifyMethod) != 0) {
+      LOG("ERROR: FramePop event is for wrong method: expected %s, got %s\n", lastNotifyMethod, name);
+      failed = JNI_TRUE;
+      fatal(jni, "DBG: FramePop event in wrong method\n");
+    }
+    deallocate(jvmti, jni, lastNotifyMethod);
+  }
+  deallocate(jvmti, jni, csig);
+  deallocate(jvmti, jni, name);
+}
+
+static
+jint Agent_Initialize(JavaVM *jvm, char *options, void *reserved) {
+  jint res;
+  jvmtiError err;
+
+  res = JNI_ENV_PTR(jvm)->GetEnv(JNI_ENV_ARG(jvm, (void **) &jvmti), JVMTI_VERSION_9);
+  if (res != JNI_OK || jvmti == NULL) {
+    LOG("GetEnv(JVMTI_VERSION_9) failedL error(%d)", res);
+    return JNI_ERR;
+  }
+  err = jvmti->GetPotentialCapabilities(&caps);
+  check_jvmti_error(err, "Agent: GetPotentialCapabilities failed");
+
+  err = jvmti->AddCapabilities(&caps);
+  check_jvmti_error(err, "Agent: AddCapabilities failed");
+
+  err = jvmti->GetCapabilities(&caps);
+  check_jvmti_error(err, "Agent: GetCapabilities failed");
+
+  if (caps.can_generate_frame_pop_events) {
+    callbacks.FramePop = &FramePop;
+    err = jvmti->SetEventCallbacks(&callbacks, sizeof(callbacks));
+    check_jvmti_error(err, "Agent: SetEventCallbacks failed");
+  }
+  return JNI_OK;
+}
+
+JNIEXPORT jboolean JNICALL
+Java_NotifyFramePopStressTest_canGenerateFramePopEvents(JNIEnv *env, jclass cls) {
+  return caps.can_generate_frame_pop_events ? JNI_TRUE : JNI_FALSE;
+}
+
+JNIEXPORT void JNICALL
+Java_NotifyFramePopStressTest_setFramePopNotificationMode(JNIEnv *env, jclass cl, jthread thread, jboolean enable) {
+  jvmtiEventMode mode = enable ? JVMTI_ENABLE : JVMTI_DISABLE;
+  set_event_notification_mode(jvmti, env, mode, JVMTI_EVENT_FRAME_POP, thread);
+}
+
+/*
+ * Call NotifyFramePop on the current frame.
+ */
+JNIEXPORT jboolean JNICALL
+Java_NotifyFramePopStressTest_notifyFramePop(JNIEnv *jni, jclass cls, jthread thread) {
+  jmethodID method;
+  jlocation loc;
+  char* name;
+  jvmtiError err;
+  jboolean isMain;
+
+  err = jvmti->GetFrameLocation(thread, 0, &method, &loc);
+  check_jvmti_status(jni, err, "notifyFramePop: Failed in JVMTI GetFrameLocation");
+
+  name = get_method_name(jvmti, jni, method);
+
+  // We only want to do a NotifyFramePop once for the main method. The sole purpose is
+  // to force the thread into interpOnly mode, which seems to help the tests's timing
+  // in a way that makes it more likely to reproduce the issue.
+  isMain = (strcmp(name, "main") == 0);
+  if (isMain) {
+    if (seenMain) {
+      return JNI_FALSE; // Only do NotifyFramePop once for main()
+    } else {
+      seenMain = JNI_TRUE;
+    }
+  }
+
+  err= jvmti->NotifyFramePop(thread, 0);
+  if (err == JVMTI_ERROR_OPAQUE_FRAME || err == JVMTI_ERROR_DUPLICATE) {
+    return JNI_FALSE;
```

Comment:
`deallocate(jvmti, jni, name);`

Reply:
Also add a LOG similar to the one below that is done if successful.

Reply:
Thanks. Added lines as suggested.